clang/tools/driver/cc1_main.cpp

```diff
@@ -407,8 +446,9 @@ int cc1_main(ArrayRef<const char *> Argv, const char *Argv0, void *MainAddr) {
     }
 
     // Set up the output backend so we can save / cache the result after.
+    auto Mapper = createCASPrefixMapperFromInvocation(Clang->getInvocation());
     CASOutputs = llvm::makeIntrusiveRefCnt<llvm::cas::CASOutputBackend>(
-        *CAS, std::move(CASIDFileOutBackend));
+        *CAS, std::move(CASIDFileOutBackend), Mapper);
```

Comment:
I think the principled approach would move code similar to this to something before `createResultCacheKey()`. Probably it needs to:
- Modify the CompilerInvocation.
- Save some side state for replaying a cache hit and/or filtering/decanonicalizing during a cache miss.
- Create a new Argv array to send to `createResultCacheKey()` based on the pruned/canonicalized CompilerInvocation.